docs/api/session.md

```diff
@@ -180,6 +180,67 @@ Emitted when a hunspell dictionary file download fails.  For details
 on the failure you should collect a netlog and inspect the download
 request.
 
+#### Event: 'select-hid-device'
+
+Returns:
+
+* `event` Event
+* `deviceList` [HIDDevice[]](structures/hid-device.md)
+* `webContents` [WebContents](web-contents.md)
+* `callback` Function
+  * `deviceId` String
+
+Emitted when a HID device needs to be selected when a call to
+`navigator.hid.requestDevice` is made. `callback` should be called with
+`deviceId` to be selected, passing an empty string to `callback` will
+cancel the request.  Additionally, permissioning on `navigator.hid` can
+be managed by using [ses.setPermissionCheckHandler(handler)](#sessetpermissioncheckhandlerhandler)
+with the `hid` permission.
+
+```javascript
+const { app, BrowserWindow } = require('electron')
+
+let win = null
+
+app.whenReady().then(() => {
+  win = new BrowserWindow({
+    width: 800,
+    height: 600
+  })
+  win.webContents.session.on('select-hid-device', (event, deviceList, webContents, callback) => {
+    event.preventDefault()
+    const selectedPort = deviceList.find((device) => {
+      return device.vendorId === '9025' && device.productId === '67'
+    })
+    if (!selectedPort) {
+      callback('')
+    } else {
+      callback(selectedPort.deviceId)
+    }
+  })
```

Comment:
Should this example include an example of using `ses.setPermissionCheckHandler` / `ses.setPermissionRequestHandler`?

Reply:
Oh good call.  I'll add that.